include/GafferImage/ImagePlug.h

```diff
@@ -166,6 +166,36 @@ class GAFFERIMAGE_API ImagePlug : public Gaffer::ValuePlug
 		};
 		//@}
 
+		/// Utility class for managing accessing plugs in the correct view
+		/// for image nodes with multiple inputs.  validViews should be set to the
+		/// viewNames() of the plug you are accessing.
+		///
+		/// If the current view in the context is valid ( either there is a view, and
+		/// it is in validViews, or validViews is empty and there is no view set ),
+		/// then ViewCheckScope does nothing, and isPresent will return true.
+		///
+		/// If there is a view set, but validViews is empty, then ViewCheckScope will
+		/// create a scope that remove view from the context, and isValid will return
+		/// true ( a viewless input should affect every view ).
+		///
+		/// If there is a view set which is not in validViews, then isPresent will return
+		/// false ( it's a valid operation, but this input does not affect this view ).
+		///
+		/// If there is no view set, but validViews is non-empty, an exception will be
+		/// thrown.  In order to use a multi view input, views must be declared on the
+		/// output.
+		struct GAFFERIMAGE_API ViewCheckScope
+		{
+			ViewCheckScope( const Gaffer::Context *context, const std::vector< std::string > &validViews );
+			bool isPresent();
+			const Gaffer::Context *context();
+		private:
+			std::unique_ptr<Gaffer::Context::EditableScope> m_editableScope;
```

Comment:
The downside is paying for the size of an EditableScope instead of just a pointer as part of ViewCheckScope in the common case where this scope doesn't get used.  But I'm OK with switching to optional if you prefer.

Reply:
Let's see how my ViewScope proposal below pans out, since that would intrinsically _be_ an EditableScope anyway.